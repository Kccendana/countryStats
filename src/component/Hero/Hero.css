section.hero {
  background-color: #fc5193;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  gap: 1em;
  height: 20%;
  overflow: hidden;
}

.hero img {
  width: 50%;
  height: auto;
  object-fit: cover;
}

.continent-detail {
  width: 50%;
  display: flex;
  align-items: flex-start;
  flex-direction: column;
}

.continent-detail h1 {
  font-size: 1.5em;
  font-weight: 700;
}

.continent-detail p {
  margin-top: -0.25em;
}
