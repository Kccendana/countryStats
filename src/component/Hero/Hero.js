import React from 'react';
import { useSelector } from 'react-redux';
import asia from '../../asset/asia.svg';
import africa from '../../asset/africa.svg';
import oceania from '../../asset/oceania.svg';
import americas from '../../asset/americas.svg';
import world from '../../asset/world.svg';
import europe from '../../asset/europe.svg';
import './Hero.css';

function Hero() {
  const { region, countries } = useSelector((state) => state.countries);

  const regionImages = {
    asia,
    africa,
    oceania,
    americas,
    europe,
    world,
  };

  return (
    <section className="hero">
      <img src={regionImages[region] || regionImages.world} alt={region} />
      <div className="continent-detail">
        <h1>{region.toUpperCase() || 'WORLD'}</h1>
        <p>
          {countries.length}
          {' '}
          countries
        </p>
      </div>
    </section>
  );
}

export default Hero;
