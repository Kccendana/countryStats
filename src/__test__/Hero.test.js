import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Hero from '../component/Hero/Hero';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('renders Hero component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Hero />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
