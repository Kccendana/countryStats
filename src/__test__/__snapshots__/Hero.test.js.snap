// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Hero component matches the snapshot 1`] = `
<div>
  <section
    class="hero"
  >
    <img
      alt=""
      src="world.svg"
    />
    <div
      class="continent-detail"
    >
      <h1>
        WORLD
      </h1>
      <p>
        0
         
        countries
      </p>
    </div>
  </section>
</div>
`;
